commit 5886fb73ba6cc13c9a634f563b9e4ea6a749509e
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date:   Wed Dec 9 13:52:31 2015 +0100

    Enforce screen locking before requesting to switch to a different user.

--- a/mate-session/gsm-manager.c
+++ b/mate-session/gsm-manager.c
@@ -3139,6 +3139,19 @@
         case GTK_RESPONSE_DELETE_EVENT:
                 break;
         case GSM_LOGOUT_RESPONSE_SWITCH_USER:
+
+                ; /* place an empty statement between label above and declaration below... */
+
+                /* Lock screen before requesting user switch
+                 */
+                GError *error;
+                error = NULL;
+                g_spawn_command_line_async ("mate-screensaver-command --lock", &error);
+                if (error != NULL) {
+                    g_warning ("Couldn't lock screen: %s", error->message);
+                    g_error_free (error);
+                }
+
                 request_switch_user (manager);
                 break;
         case GSM_LOGOUT_RESPONSE_HIBERNATE:
